### ISSUE TITLE
Using Gradient Descent on mnist

### ISSUE THREAD

**Issue by ati6868** [BindsNET/bindsnet#346]
Hi
i need to use Gradient Descent to update weights.
In here, first we count the number of generated spikes by layer `B` and then i compare generated spikes(firing_rate) with image pixels and difference is loss as the defined below:

`spikes = {
    "A": Input_monitor.get("s"), "B": LIF_monitor.get("s") 
}`
`Generated_spikes=spikes["B"]`
`sum_Generated_spikes =(Generated_spikes==1).sum(dim=0) #counts number of 1's `
`loss = sum_Generated_spikes - images[0]`

Now i need to use GD to update weights ...i used this code
`optimizer.zero_grad()`
` loss.backward()`
`optimizer.step()`
i set attributes of parameters like weights to  `requires_grad=True` but where i should use a loop to do it on whole of my training examples?
Could someone please help me on this??
